model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+                    logger.severe(e.toString());
+                    throw e;
+                }
+            }
+            // Add on its label
+            try{
+                MetaItem<String> label = new MetaItem<>(MetaItemType.LABEL);
+                label.setValue(parameter.propertyName);
+                metaMap.add(label);
+            }catch (Exception e){
+                logger.severe(parameter.propertyName + "Failed label");
+                logger.severe(e.toString());
+                throw e;
+            }
+            //Use the MetaMap to create an AttributeDescriptor
+            AttributeDescriptor<?> attributeDescriptor = new AttributeDescriptor<>(parameter.propertyId.toString(), attributeType, metaMap);
+
+
+
+            //Use the AttributeDescriptor and the Value to create a new Attribute
+            Attribute<?> generatedAttribute = new Attribute(attributeDescriptor, value.get());
+            // Add it to the AttributeMap
+            attributes.addOrReplace(generatedAttribute);
+        }
+        //Special parameter definitions without being defined in the AVL Parameter List, thanks Teltonika
+
+        //latlng are the latitude-longitude coordinates, also check if it's 0,0, if it is, don't update.
+        if (reported.containsKey("latlng") && !Objects.equals(reported.get("latlng"), "0.000000,0.000000")){
+            try{
+                String latlngString = reported.get("latlng").toString();
+                GeoJSONPoint point = ParseLatLngToGeoJSONObject(latlngString);
+                Attribute<?> attr = new Attribute<>("location", ValueType.GEO_JSON_POINT, point);
+
+                attributes.add(attr);
+            }catch (Exception e){
+                logger.severe("Failed coordinates");
+                logger.severe(e.toString());
+                throw e;
+            }
+        }
+        //Timestamp grabbed from the device.
+        //TODO: It doesn't look like the timestamp is being parsed correctly...
+        if (reported.containsKey("ts")){
+            try{
+                long unixTimestampMillis = Long.parseLong(reported.get("ts").toString());
+                Timestamp deviceTimestamp = Timestamp.from(Instant.ofEpochMilli(unixTimestampMillis));
+                attributes.add(new Attribute<>("lastContact", ValueType.DATE_AND_TIME, deviceTimestamp).setTimestamp(deviceTimestamp.getTime()));
+
+
+                //Update all affected attribute timestamps
+                attributes.forEach(attribute -> attribute.setTimestamp(deviceTimestamp.getTime()));
+            }catch (Exception e){
+                logger.severe("Failed timestamps");
+                logger.severe(e.toString());
+                throw e;
+            }
+
+        }
+
+        // Store data points, allow use for rules, and don't allow user parameter modification, for every attribute parsed
+        try{
+            attributes.forEach(attribute -> attribute.addOrReplaceMeta(
```

Comment:
Maybe this should only be done for new attributes so user can potentially decide to not store data points for an attribute by removing the configuration item

Reply:
Actually, by relating this to https://github.com/openremote/openremote/pull/1147#discussion_r1402049387, should I create some sort of mediary function that checks the existing attribute's attributes against the parsed ones, and then try to merge all of the metadata of the attribute? I have considered this before, but this requires a conversation of what we consider to be more important, the current attributes of the asset or the incoming ones.